activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1915,7 +1915,14 @@ def build_with_expression_from_value(value)
         when Arel::Nodes::SqlLiteral then Arel::Nodes::Grouping.new(value)
         when ActiveRecord::Relation then value.arel
         when Arel::SelectManager then value
-        when Array then value.map { |q| build_with_expression_from_value(q) }.reduce { |result, value| result.union(:all, value) }
+        when Array
+          parts = value.map { |q| build_with_expression_from_value(q).ast }
```

Comment:
This will fail on `.ast` not defined when we pass non AR::Relation in the array. For example, line 1915 - `Arel::Nodes::SqlLiteral` instances do not respond to `.to_ast`

Reply:
We probably ought to add a test for that then.

Reply:
Updated the test case to also test this.